x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/common/Truncator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.common;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Provides truncation logic for inference requests
+ */
+public class Truncator {
+
+    /**
+     * Defines the percentage to reduce the input text for an inference request.
+     */
+    static final Setting<Double> REDUCTION_PERCENTAGE_SETTING = Setting.doubleSetting(
+        "xpack.inference.truncator.reducation_percentage",
+        0.5,
+        0.0,
+        1.0,
```

Comment:
Should we limit this to `0.9999` or something? No one should be using `1.0` here because it'll just return the text as it is. Same idea with `0.0`.

Reply:
> Should we limit this to 0.9999 or something?

How about 0.99 and 0.01?